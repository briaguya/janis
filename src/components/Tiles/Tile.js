import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { misc } from 'js/i18n/definitions';
import { tilePropTypes } from './proptypes';

import guideIcon from 'images/guide_icon.png';

const Tile = ({ url, text, compact, pageName }) => {
  let typeIcon
  if (pageName == "GuidePageNode") {
    typeIcon = guideIcon
  }
  return (
    <Link className={compact ? 'coa-Tile--compact' : 'coa-Tile'} to={url}>
      <div className="coa-Tile__content">
        <p className="coa-Tile__text">{text}</p>
        <i className="material-icons">
          {typeIcon
            ? (<img src={typeIcon} alt={misc.guideIcon.defaultMessage} />)
            : "arrow_forward"
          }
        </i>
      </div>
    </Link>
  )
};

Tile.propTypes = tilePropTypes;

export default Tile;
