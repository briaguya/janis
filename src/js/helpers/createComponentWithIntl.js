// this is a helper function for the jest snapshot tests
// https://github.com/formatjs/react-intl/blob/master/docs/Testing-with-React-Intl.md#snapshot-testing

import React from 'react';
import renderer from 'react-test-renderer';
import {IntlProvider} from 'react-intl';

const createComponentWithIntl = (children, props = {locale: 'en'}) => {
  return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};

export default createComponentWithIntl;
