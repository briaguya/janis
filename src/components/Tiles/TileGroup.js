import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { services as i18n3 } from 'js/i18n/definitions';
import I18nLink from 'components/I18n/I18nLink';
import ArrowRight from 'components/SVGs/ArrowRight';

import Tile from './Tile';
import { tileGroupPropTypes } from './proptypes';

const TileGroup = ({ tiles, text, url, description, contentType, intl }) => (
  <div
    className={`coa-TileGroup ${
      contentType === intl.formatMessage(i18n3.service)
        ? 'coa-TileGroup--service'
        : ''
    }`}
  >
    <div className="wrapper container-fluid">
      {text &&
        url && (
          <h4 className="coa-TileGroup__title">
            <I18nLink to={url}>
              {text}&nbsp;<ArrowRight />
            </I18nLink>
          </h4>
        )}

      {text && !url && <h4 className="coa-TileGroup__title">{text}</h4>}

      <div className="row">
        {tiles.map(({ url, text }, index) => (
          <div
            key={index}
            className="coa-TileGroup__tile col-xs-12 col-md-6 col-lg-3"
          >
            <Tile url={url} text={text} />
          </div>
        ))}
      </div>
    </div>
  </div>
);

TileGroup.propTypes = tileGroupPropTypes;

export default injectIntl(TileGroup);
