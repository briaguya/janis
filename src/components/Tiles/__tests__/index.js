import React from 'react';
import renderer from 'react-test-renderer';
// BrowserRouter is necessary so the tests know where to find Link
import { BrowserRouter } from 'react-router-dom';

import createComponentWithIntl from 'js/helpers/createComponentWithIntl';
import Tile from './../Tile';
import TileGroup from './../TileGroup';

describe('Tile', () => {
  test('tile renders compact true', () => {
    const component = renderer.create(
      <BrowserRouter>
      	<Tile
      		compact={true}
          text="home"
          url="https://alpha.austin.gov"
      	/>
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('tile renders with compact false', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Tile
          compact={false}
          text="home"
          url="https://alpha.austin.gov"
        />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});

describe('TileGroup', () => {
  test('tile group renders with compact true', () => {
    const tiles = [
      {
        "url":"/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/",
        "title":"Set out your leaves, grass, and branches for pickup"
      },
      {
        "url":"/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/",
        "title":"Set out your leaves, grass, and branches for pickup"
      },
    ];
    const intl = {"locale": "en"};

    const component = createComponentWithIntl(
      <BrowserRouter>
        <TileGroup
          title="Chronic disease prevention →"
          titleUrl="/health-safety/disaster-safety-relief/chronic-disease-prevention"
          description="Chronic diseases are ongoing medical conditions such as diabetes..."
          tiles={tiles}
          compact
          intl={intl}
        />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('tile group renders with compact false', () => {
    const tiles = [
      {
        "url":"/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/",
        "title":"Set out your leaves, grass, and branches for pickup"
      },
      {
        "url":"/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/",
        "title":"Set out your leaves, grass, and branches for pickup"
      },
    ];
    const intl = {"locale": "en"};

    const component = createComponentWithIntl(
      <BrowserRouter>
        <TileGroup
          title="Chronic disease prevention →"
          titleUrl="/health-safety/disaster-safety-relief/chronic-disease-prevention"
          description="Chronic diseases are ongoing medical conditions such as diabetes..."
          tiles={tiles}
          compact={false}
          intl={intl}
        />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
