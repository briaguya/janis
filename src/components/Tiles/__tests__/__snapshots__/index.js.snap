// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile tile renders compact true 1`] = `
<a
  className="coa-Tile--compact"
  href="/https://alpha.austin.gov"
  onClick={[Function]}
>
  <div
    className="coa-Tile__content"
  >
    <p
      className="coa-Tile__text"
    >
      home
    </p>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </div>
</a>
`;

exports[`Tile tile renders with compact false 1`] = `
<a
  className="coa-Tile"
  href="/https://alpha.austin.gov"
  onClick={[Function]}
>
  <div
    className="coa-Tile__content"
  >
    <p
      className="coa-Tile__text"
    >
      home
    </p>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </div>
</a>
`;

exports[`TileGroup tile group renders with compact false 1`] = `
<div
  className="coa-TileGroup"
>
  <h4
    className="coa-TileGroup__title"
  >
    <a
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention"
      onClick={[Function]}
    >
      Chronic disease prevention →
    </a>
  </h4>
  <p
    className="coa-TileGroup__description"
  >
    Chronic diseases are ongoing medical conditions such as diabetes...
  </p>
  <div
    className="coa-TileGroup__tiles-container"
  >
    <a
      className="coa-Tile"
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/"
      onClick={[Function]}
    >
      <div
        className="coa-Tile__content"
      >
        <p
          className="coa-Tile__text"
        >
          Set out your leaves, grass, and branches for pickup
        </p>
        <i
          className="material-icons"
        >
          arrow_forward
        </i>
      </div>
    </a>
    <a
      className="coa-Tile"
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/"
      onClick={[Function]}
    >
      <div
        className="coa-Tile__content"
      >
        <p
          className="coa-Tile__text"
        >
          Set out your leaves, grass, and branches for pickup
        </p>
        <i
          className="material-icons"
        >
          arrow_forward
        </i>
      </div>
    </a>
  </div>
</div>
`;

exports[`TileGroup tile group renders with compact true 1`] = `
<div
  className="coa-TileGroup coa-TileGroup--compact"
>
  <h4
    className="coa-TileGroup__title"
  >
    <a
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention"
      onClick={[Function]}
    >
      Chronic disease prevention →
    </a>
  </h4>
  <p
    className="coa-TileGroup__description"
  >
    Chronic diseases are ongoing medical conditions such as diabetes...
  </p>
  <div
    className="coa-TileGroup__tiles-container--compact"
  >
    <a
      className="coa-Tile--compact"
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/"
      onClick={[Function]}
    >
      <div
        className="coa-Tile__content"
      >
        <p
          className="coa-Tile__text"
        >
          Set out your leaves, grass, and branches for pickup
        </p>
        <i
          className="material-icons"
        >
          arrow_forward
        </i>
      </div>
    </a>
    <a
      className="coa-Tile--compact"
      href="/en/health-safety/disaster-safety-relief/chronic-disease-prevention/set-out-your-leaves-grass-and-branches-for-pickup/"
      onClick={[Function]}
    >
      <div
        className="coa-Tile__content"
      >
        <p
          className="coa-Tile__text"
        >
          Set out your leaves, grass, and branches for pickup
        </p>
        <i
          className="material-icons"
        >
          arrow_forward
        </i>
      </div>
    </a>
  </div>
</div>
`;
